application/src/main/java/org/togetherjava/tjbot/moderation/ModAuditLogWriter.java

```diff
@@ -0,0 +1,152 @@
+     *
+     * @param timestamp the timestamp of the action. usually use
+     *        {@link GenericInteractionCreateEvent#getTimeCreated()}
+     *
+     * @param guild the guild. usually use {@link GenericInteractionCreateEvent#getGuild()}
+     *
+     * @param attachments the attachments that'll be added to the message
+     */
+    public static void writeModAuditLog(@NotNull String title, @NotNull String description,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull Guild guild,
+            @NotNull List<@NotNull Attachment> attachments) {
+        Optional<TextChannel> auditLogChannel = getModAuditLogChannel(guild);
+        if (auditLogChannel.isEmpty()) {
+            return;
+        }
+
+        MessageAction message = auditLogChannel.orElseThrow()
+            .sendMessageEmbeds(new EmbedBuilder().setTitle(title)
+                .setDescription(description)
+                .setAuthor(author.getAsTag(), null, author.getAvatarUrl())
+                .setTimestamp(timestamp)
+                .setColor(EMBED_COLOR)
+                .build());
+
+        for (Attachment attachment : attachments) {
+            message = message.addFile(attachment.getContent(), attachment.getName());
+        }
+        message.queue();
+    }
+
+    /**
+     * logs an entry in the mod audit log channel.
+     *
+     * @param title the title of the log embed
+     *
+     * @param description the description of the log embed
+     *
+     * @param author the user who triggered the action
+     *
+     * @param timestamp the timestamp of the action. usually use
+     *        {@link GenericInteractionCreateEvent#getTimeCreated()}
+     *
+     * @param guild the guild. usually use {@link GenericInteractionCreateEvent#getGuild()}
+     *
+     * @param attachment an attachment that'll be added to the message
+     */
+    public static void writeModAuditLog(@NotNull String title, @NotNull String description,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull Guild guild,
+            @NotNull Attachment attachment) {
+        writeModAuditLog(title, description, author, timestamp, guild, List.of(attachment));
+    }
+
+    /**
+     * logs an entry in the mod audit log channel.
+     *
+     * @param title the title of the log embed
+     *
+     * @param description the description of the log embed
+     *
+     * @param author the user who triggered the action
+     *
+     * @param timestamp the timestamp of the action. usually use
+     *        {@link GenericInteractionCreateEvent#getTimeCreated()}
+     *
+     * @param guild the guild. usually use {@link GenericInteractionCreateEvent#getGuild()}
+     */
+    public static void writeModAuditLog(@NotNull String title, @NotNull String description,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull Guild guild) {
+        writeModAuditLog(title, description, author, timestamp, guild, List.of());
+    }
```

Comment:
NIT: I think those overloads are obsolete and rather confusing. I would remove them.

Reply:
I've added overloads to the method, so we have one that takes a `List`, one that takes a single `Attachment`, and one that takes non. I don't think it's necessary.

Reply:
Yeah I saw that but you can also just only offer the list overload and they can all use that instead? Dunno, I dont have a strong opinion on this but I think I would prefer it without the overloads.

If you want to keep them however, you have to outline the difference in the Javadoc.

Reply:
see e52d974401ccc7be9b1c7de0ede6a270b7ab2d54